src/WebJobs.Script/WorkerUtilities.cs

```diff
@@ -17,5 +17,41 @@ public static int GetUnusedTcpPort()
                 return port;
             }
         }
+
+        /// <summary>
+        /// Determines whether the specified port is available.
+        /// </summary>
+        internal static bool CanBindToPort(int port)
+        {
+            // Try to bind to the port using IPv6 dual mode socket to cover both IPv4 and IPv6.
+            using (var tcpSocket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp) { DualMode = true })
+            {
+                try
+                {
+                    tcpSocket.Bind(new IPEndPoint(IPAddress.IPv6Any, port));
+                    return true;
+                }
+                catch (SocketException se) when (se.SocketErrorCode == SocketError.AddressAlreadyInUse)
+                {
+                    return false;
+                }
+                catch
+                {
+                    // Fall back to IPv4 only socket if IPv6 is not supported on the platform.
+                    using (var tcpSocketAny = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
```

Comment:
Use the new using style:


```suggestion
                    using Socket tcpSocketAny = new(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```